File: 3940-limit-occurrences-in-sorted-array/3940-limit-occurrences-in-sorted-array.cpp
class Solution {
public:
    vector<int> limitOccurrences(vector<int>& nums, int k) {
        vector<int>r;
        int c=0;
        int prev=INT_MIN;
        for(int i=0;i<nums.size();i++){
            if(nums[i]!=prev){
                prev=nums[i];
                c=1;
            }
            else{
                c++;
            }
            if(c<=k)r.push_back(nums[i]);
        }
        return r;
    }
};

File: 3940-limit-occurrences-in-sorted-array/3940-limit-occurrences-in-sorted-array.java
class Solution {
    public int[] limitOccurrences(int[] nums, int k) {
        List<Integer> s=new ArrayList<>();
        int prev=Integer.MIN_VALUE;
        int c=0;
        for(int i=0;i<nums.length;i++){
            if(nums[i]!=prev){
                c=1;
                prev = nums[i];
            }
            else{
                c++;
            }
            if(c<=k){
                s.add(nums[i]);
            }
        }
        int[] result = new int[s.size()];
        for (int i = 0; i < s.size(); i++) {
            result[i] = s.get(i);
        }
        return result;
    }
}